Lib/contextlib.py

```diff
@@ -303,6 +303,31 @@ def __exit__(self, *exc_info):
         self.thing.close()
 
 
+class aclosing(AbstractAsyncContextManager):
+    """Async context manager for safely finalizing an async generator using its ``aclose()`` method.
+
+    Code like this:
+
+        async with aclosing(<module>.fetch(<arguments>)) as agen:
+            <block>
+
+    is equivalent to this:
+
+        agen = await <module>.fetch(<arguments>)
```

Comment:
Sorry, I said the `await` should be here, but that was incorrect.

The object passed to `aclosing()` is just taken as is, returned by `__aenter__`, and has `aclose()` called in `__aexit__`.

Reply:
Oh, I also missed to check it. Reverted here.